budget-service/src/main/java/org/java/mentorship/budget/controller/TransactionController.java

```diff
@@ -0,0 +1,65 @@
+package org.java.mentorship.budget.controller;
+
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.java.mentorship.budget.domain.TransactionEntity;
+import org.java.mentorship.budget.domain.mapper.TransactionContractMapper;
+import org.java.mentorship.budget.service.TransactionService;
+import org.java.mentorship.contracts.budget.dto.Transaction;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/transactions")
+public class TransactionController {
+
+    private final TransactionService transactionService;
+
+    @GetMapping
+    public ResponseEntity<List<Transaction>> getAllTransactions() {
+        List<TransactionEntity> transactionEntities = transactionService.findAll();
```

Comment:
No need for this local variable. You can just do:
List<Transaction> transactions = transactionService.findAll()
                .stream()
                .map(TransactionContractMapper::entityToContract)
                .collect(Collectors.toList());